src/main/scala/com/exasol/cloudetl/emitter/FilesMetadataEmitter.scala

```diff
@@ -0,0 +1,93 @@
+package com.exasol.cloudetl.emitter
+
+import java.lang.Long
+
+import scala.collection.mutable.ListBuffer
+
+import com.exasol.ExaIterator
+import com.exasol.cloudetl.bucket.Bucket
+import com.exasol.cloudetl.storage.FileFormat
+import com.exasol.cloudetl.storage.StorageProperties
+import com.exasol.parquetio.splitter.ParquetFileSplitter
+
+import com.typesafe.scalalogging.LazyLogging
+import org.apache.parquet.hadoop.util.HadoopInputFile
+
+/**
+ * A class that calculates and emits file based metadata from a storage path.
+ *
+ * @param properties a storage properties to create configurations
+ * @param parallelism a parallelism number to create groups
+ */
+final case class FilesMetadataEmitter(properties: StorageProperties, parallelism: Int)
+    extends Emitter
+    with LazyLogging {
+
+  private[this] val bucket = Bucket(properties)
+  private[this] val paths = bucket.getPaths()
+  private[this] val fileFormat = properties.getFileFormat()
+
+  override def emit(context: ExaIterator): Unit = {
+    logger.info(s"Found total of '${paths.size}' files in path '${bucket.bucketPath}'.")
+    if (fileFormat != FileFormat.PARQUET) {
+      emitRegularFilesMetadata(context)
+    } else {
+      emitParquetFilesMetadata(context)
+    }
+  }
+
+  private[this] def emitRegularFilesMetadata(context: ExaIterator): Unit = {
+    var index = 0L
+    paths.foreach { case filename =>
```

Comment:
only one case statement? Do you need a default here?

Reply:
No, not really. This is maybe one of design flaws in the language. Since it is iterating over single values `.foreach { value =>` works, but if it were complex for example map, I need to use `map.foreach { case (key, value) =>` case.

It is good to not mix them, I removed case from single values.